ee/server/service/software_installers.go

```diff
@@ -393,6 +398,11 @@ func (svc *Service) UpdateSoftwareInstaller(ctx context.Context, payload *fleet.
 		if uninstallScript == "" { // extension can't change on an edit so we can generate off of the existing file
 			uninstallScript = file.GetUninstallScript(existingInstaller.Extension)
 		}
+		if uninstallScript == "" {
+			return nil, &fleet.BadRequestError{
+				Message: fmt.Sprintf("Couldn't edit. Uninstall script is required for .%s packages.", strings.ToLower(existingInstaller.Extension)),
+			}
+		}
```

Comment:
I see that we are checking for empty install and un-install scripts here. But the ticket refers to "on exe uploads". Curious how this check is scoped it exe files.

Reply:
So, on edit we let people clear out scripts to put them back at defaults for a given installer type, and since the installer type is locked once the package is uploaded we can use the old installer type to determine which script to use.

The exception to this is EXE of course, where we fail because we don't allow empty install scripts.

This setup works because right above is (and for the install script) we set the script to the default for the extension if nothing is provided. For EXEs, we'll get a blank script back (and fail here). For everything else, we'll get a default script back that isn't blank (and won't fail here).